opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java

```diff
@@ -0,0 +1,213 @@
+    if (arguments.get("time_field").getValue() == null) {
+      rcfType = FunctionName.BATCH_RCF;
+      return BatchRCFParams.builder()
+              .shingleSize((Integer) arguments.get("shingle_size").getValue())
+              .build();
+    }
+    rcfType = FunctionName.FIT_RCF;
+    return FitRCFParams.builder()
+            .shingleSize((Integer) arguments.get("shingle_size").getValue())
+            .timeDecay((Double) arguments.get("time_decay").getValue())
+            .timeField((String) arguments.get("time_field").getValue())
+            .dateFormat("yyyy-MM-dd HH:mm:ss")
+            .build();
+  }
+
+  private Map<String, ExprValue> convertRowIntoExprValue(ColumnMeta[] columnMetas, Row row) {
+    ImmutableMap.Builder<String, ExprValue> resultBuilder = new ImmutableMap.Builder<>();
+    for (int i = 0; i < columnMetas.length; i++) {
+      ColumnValue columnValue = row.getValue(i);
+      String resultKeyName = columnMetas[i].getName();
+      popluateResultBuilder(columnValue, resultKeyName, resultBuilder);
+    }
+    return resultBuilder.build();
+  }
+
+  private void popluateResultBuilder(ColumnValue columnValue,
+                          String resultKeyName,
+                          ImmutableMap.Builder<String, ExprValue> resultBuilder) {
+    switch (columnValue.columnType()) {
+      case INTEGER:
+        resultBuilder.put(resultKeyName, new ExprIntegerValue(columnValue.intValue()));
+        break;
+      case DOUBLE:
+        resultBuilder.put(resultKeyName, new ExprDoubleValue(columnValue.doubleValue()));
+        break;
+      case STRING:
+        resultBuilder.put(resultKeyName, new ExprStringValue(columnValue.stringValue()));
+        break;
+      case SHORT:
+        resultBuilder.put(resultKeyName, new ExprShortValue(columnValue.shortValue()));
+        break;
+      case LONG:
+        resultBuilder.put(resultKeyName, new ExprLongValue(columnValue.longValue()));
+        break;
+      case FLOAT:
+        resultBuilder.put(resultKeyName, new ExprFloatValue(columnValue.floatValue()));
+        break;
+      case BOOLEAN:
+        resultBuilder.put(resultKeyName, new ExprBooleanValue(columnValue.booleanValue()));
+        break;
+      default:
+        break;
+    }
+  }
+
+  private Map<String, ExprValue> convertResultRowIntoExprValue(ColumnMeta[] columnMetas,
+                                                               Row row,
+                                                               Map<String, ExprValue> schema) {
+    ImmutableMap.Builder<String, ExprValue> resultBuilder = new ImmutableMap.Builder<>();
+    for (int i = 0; i < columnMetas.length; i++) {
+      ColumnValue columnValue = row.getValue(i);
+      String resultKeyName = columnMetas[i].getName();
+      // change key name to avoid duplicate key issue in result map
+      // only value will be shown in the final returned result
+      if (schema.containsKey(resultKeyName)) {
+        resultKeyName = resultKeyName + "1";
```

Comment:
Is it general name convention in SQL to append "1" to avoid duplicate name?

Reply:
Not sure what's the best practice to address this, need to discuss it with package owner to find out. All the other command arguments are built into a list, instead of a map that we used here. So no similar case to reference.